modules/dxp/apps/analytics/analytics-reports-web/src/main/java/com/liferay/analytics/reports/web/internal/model/PaidTrafficChannel.java

```diff
@@ -0,0 +1,160 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * The contents of this file are subject to the terms of the Liferay Enterprise
+ * Subscription License ("License"). You may not use this file except in
+ * compliance with the License. You can obtain a copy of the License by
+ * contacting Liferay, Inc. See the License for the specific language governing
+ * permissions and limitations under the License, including but not limited to
+ * distribution rights of the Software.
+ *
+ *
+ *
+ */
+
+package com.liferay.analytics.reports.web.internal.model;
+
+import com.liferay.analytics.reports.web.internal.model.util.TrafficChannelUtil;
+import com.liferay.portal.kernel.json.JSONArray;
+import com.liferay.portal.kernel.json.JSONObject;
+import com.liferay.portal.kernel.json.JSONUtil;
+import com.liferay.portal.kernel.util.ListUtil;
+import com.liferay.portal.kernel.util.ResourceBundleUtil;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.Locale;
+import java.util.Objects;
+import java.util.ResourceBundle;
+import java.util.stream.Stream;
+
+/**
+ * @author David Arques
+ */
+public class PaidTrafficChannel implements TrafficChannel {
```

Comment:
All this classes should end with the prefix Impl.

For example PaidTrafficChannelImpl

Reply:
Uhm, but I've seen other examples following this pattern:

Here, for example:
https://github.com/liferay/liferay-portal/blob/master/modules/apps/content-dashboard/content-dashboard-journal-impl/src/main/java/com/liferay/content/dashboard/journal/internal/item/action/EditJournalArticleContentDashboardItemAction.java

Reply:
I think Action is a different pattern.

Reply:
I've seen that in general when there's just one implementation the suffix _-Impl_ is used but when there's more than one the _-Impl_ is not necesary. Take into account we don't have a `TrafficChannelImpl`. But, if you still think that adding a suffix it's better I'm ok with this.

More examples:

- https://github.com/liferay/liferay-portal/blob/master/modules/apps/content-dashboard/content-dashboard-web/src/main/java/com/liferay/content/dashboard/web/internal/item/type/DDMStructureContentDashboardItemType.java

- https://github.com/liferay/liferay-portal/blob/master/modules/apps/content-dashboard/content-dashboard-web/src/main/java/com/liferay/content/dashboard/web/internal/item/JournalArticleContentDashboardItem.java

Reply:
suffix added ;)